test/ui/segmented-bar/segmented-bar-spec.js

```diff
@@ -0,0 +1,110 @@
+                    segmentedBarComponent.prepareForActivationEvents();
+                    segmentedBarComponent.templateObjects.repetition.selection = [1];
+                    expect(callback).toHaveBeenCalled();
+                    // teardown to not affect future tests
+                    segmentedBarComponent.exitDocument();
+                });
+
+                it("should dispatchActionEvent with correct .detail data", function() {
+                    segmentedBarComponent.addEventListener("action", callback, false);
+                    segmentedBarComponent.prepareForActivationEvents();
+                    segmentedBarComponent.templateObjects.repetition.selection = [1];
+                    expect(callback).toHaveBeenCalled();
+                    // teardown to not affect future tests
+                    segmentedBarComponent.exitDocument();
+                });
+
+                it("should not dispatchActionEvent when isSelectionEnabled === false", function() {
+                    segmentedBarComponent.isSelectionEnabled = false;
+                    segmentedBarComponent.addEventListener("action", callback, false);
+                    segmentedBarComponent.templateObjects.repetition.selection = [1];
+                    expect(callback).not.toHaveBeenCalled();
+                    // teardown to not affect future tests
+                    segmentedBarComponent.exitDocument();
+                });
+            });
+        });
+
+        describe("Segment", function () {
+            it("should instantiate", function () {
+                expect(segmentElements[0].component).toBeDefined();
+            });
+
+            it("should render with correct text label from content array", function () {
+                expect(segmentElements[3].component.templateObjects.text.element.textContent).toEqual("region4");
+            });
+
+            it("should render with correct text label from RangeController", function () {
+                segmentElements = testPage.querySelectorAll("[data-montage-id=segment2]");
+                expect(segmentElements[3].component.templateObjects.text.element.textContent).toEqual("region4");
+            });
+
+            it("should render with correct text label from TreeController", function () {
+                segmentElements = testPage.querySelectorAll("[data-montage-id=segment3]");
+                expect(segmentElements[0].component.templateObjects.text.element.textContent).toEqual("USA");
+            });
+
+            it("should render with corrent width in horizontal orientation", function () {
+                expect(segmentElements[0].clientWidth).toEqual(96);
+            });
+
+            it("should render with corrent height in vertical orientation", function () {
+                segmentElements = testPage.querySelectorAll("[data-montage-id=segment4]");
+                expect(segmentElements[0].clientHeight).toEqual(96);
+            });
+        });
+    });
+});
```

Comment:
nit: The `// teardown to not affect future tests` comments look funky. They're a little long and a little too much on their own line to be sentence fragment comments without capitalization and periods. Perhaps `// Tear down to not affect future tests.` (full sentence on a line by itself, with correct punctuation) or `segmentedBarComponent.exitDocument(); // teardown` (dependent clause describing the line it's on).